### PR TITLE
add default wait timeout for scale operation

### PR DESCRIPTION
**Pull request by ravilr**
@jimmidyson

### CODE REVIEW

**Comment by ravilr** [comment]
should it throw a kubernetesClientException if the scale wait times out?

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java [hunk 1]
```diff
@@ -99,6 +101,7 @@
   private int connectionTimeout = 10 * 1000;
   private int requestTimeout = 10 * 1000;
   private long rollingTimeout = DEFAULT_ROLLING_TIMEOUT;
+  private long scaleTimeoutInMillis = DEFAULT_SCALE_TIMEOUT;
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java]: `scaleTimeout` to be consistent with `rollingTimeout` please.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java [hunk 2]
```diff
@@ -62,11 +62,11 @@
...
+    this(client, config, null, namespace, null, true, null, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>(), false, config.getRollingTimeout(), TimeUnit.MILLISECONDS);
   }
 
   public ReplicationControllerOperationsImpl(OkHttpClient client, Config config, String apiVersion, String namespace, String name, Boolean cascading, ReplicationController item, String resourceVersion, Boolean reloadingFromServer, long gracePeriodSeconds, Map<String, String> labels, Map<String, String> labelsNot, Map<String, String[]> labelsIn, Map<String, String[]> labelsNotIn, Map<String, String> fields) {
-    this(client, config, apiVersion, namespace, name, cascading, item, resourceVersion, reloadingFromServer, gracePeriodSeconds, labels, labelsNot, labelsIn, labelsNotIn, fields, false, config.getRollingTimeout(), TimeUnit.MINUTES);
+    this(client, config, apiVersion, namespace, name, cascading, item, resourceVersion, reloadingFromServer, gracePeriodSeconds, labels, labelsNot, labelsIn, labelsNotIn, fields, false, config.getRollingTimeout(), TimeUnit.MILLISECONDS);
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java]: I assume this is to fix a different bug? LGTM :)

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java [hunk 3]
```diff
@@ -58,6 +58,7 @@
   public static final String KUBERNETES_REQUEST_TIMEOUT_SYSTEM_PROPERTY = "kubernetes.request.timeout";
   public static final String KUBERNETES_ROLLING_TIMEOUT_SYSTEM_PROPERTY = "kubernetes.rolling.timeout";
   public static final String KUBERNETES_LOGGING_INTERVAL_SYSTEM_PROPERTY = "kubernetes.logging.interval";
+  public static final String KUBERNETES_SCALE_TIMEOUT_SYSTEM_PROPERTY = "kubernetes.scale.timeoutInMillis";
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java]: `kubernetes.scale.timeout` for consistency please.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java [hunk 4]
```diff
@@ -505,6 +514,14 @@ public void setRollingTimeout(long rollingTimeout) {
     this.rollingTimeout = rollingTimeout;
   }
 
+  public long getScaleTimeoutInMillis() {
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java]: `getScaleTimeout`

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java [hunk 5]
```diff
@@ -505,6 +514,14 @@ public void setRollingTimeout(long rollingTimeout) {
...
+    return scaleTimeoutInMillis;
+  }
+
+  public void setScaleTimeoutInMillis(long scaleTimeoutInMillis) {
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java]: `setScaleTimeout`

**Comment by jimmidyson** [comment]
> should it throw a kubernetesClientException if the scale wait times out?

We don't do for rolling updates & I'd like to be consistent, but it would make more sense to do so. For this PR, let's not throw it & think about changing that in a separate PR?

**Comment by jimmidyson** [comment]
Would also be good to make this Timeoutable (supporting `withTimeout` on individual scale operations) like rolling update. If you can include it in this PR then great, otherwise we can do it in a separate PR.

Overall LGTM with minor nits on naming. Thanks @ravilr!

**Comment by jimmidyson** [comment]
ok to test

**Comment by jimmidyson** [comment]
Sorry needs a rebase since I updated the deployment reaping logic.

**Comment by ravilr** [comment]
@jimmidyson PTAL. rebased and updated to `scaleTimeout`.

**Comment by jimmidyson** [comment]
LGTM - thanks! [merge]

**Comment by jimmidyson** [comment]
FYI just releasing 1.3.94 with this & other relevant deployment fixes in - should be available in < 30 minutes.

**Pull request merged**